### PR TITLE
add some new read-only functions to market and pool

### PR DESCRIPTION
**Pull request by dbeal-eth**
should make it much easier to understand the underlying computations that go behind capacity locked or similar issues in the future

also fixes credit capacity locking with deposited collateral issue.

### CODE REVIEW

**Review by noisekit — APPROVED** [review]
can we have some tests maybe?

**Comment by dbeal-eth** [comment]
> can we have some tests maybe?

they are forthcoming tomorrow, now that fix is in hand

File: protocol/synthetix/contracts/storage/Market.sol [hunk 1]
```diff
@@ -298,7 +298,14 @@ library Market {
      *
      */
     function isCapacityLocked(Data storage self) internal view returns (bool) {
-        return self.creditCapacityD18 < getLockedCreditCapacity(self).toInt();
+        (
```
[the remark this answers is not in the document]
Comment (noisekit) [reply protocol/synthetix/contracts/storage/Market.sol:301]: As this is an internal function we need to check which public methods get affected and start requiring offchain price updates attached. May break a lot of integrations.

[the remark this answers is not in the document]
Comment (noisekit) [reply protocol/synthetix/contracts/storage/Market.sol:301]: I recommend removing the fix from `add some new read-only functions to market and pool` PR and making another one specifically for this fix.

**Review by noisekit — APPROVED** [review]

**Comment by dbeal-eth** [comment]
> > * MarketManagerModule.isMarketCapacityLocked
> 
> Used in liquidity ui

lol ok well guess we need to add erc7412 to the liquidity ui for that call (really should be wrapped around all calls imo)

**Comment by noisekit** [comment]
No

> really should be wrapped around all calls imo

Bad idea b cause that call is much heavier and harder to debug than plain call.

**Pull request merged**